### ISSUE TITLE
Build failure with /p:DebugType=embedded

### ISSUE THREAD

**Issue by L2** [dotnet/runtime#56998]
### Description

Steps to reproduce:
```
git clone https://github.com/dotnet/runtime.git net6_debugtypeembedded
cd net6_debugtypeembedded
.\build.cmd clr+libs -c Release /p:DebugType=embedded
```

Right near the end of the build it gives this error:

```
  Microsoft.Extensions.Hosting.Abstractions -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Hosting.Abstractions\netstandard2.1-Release\Microsoft.Extensions.Hosting.Abstractions.dll
  Microsoft.Extensions.Configuration.Json -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Configuration.Json\netstandard2.1-Release\Microsoft.Extensions.Configuration.Json.dll
  Microsoft.Extensions.Configuration.EnvironmentVariables -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\netstandard2.0-Release\Microsoft.Extensions.Configuration.EnvironmentVariables.dll
  Microsoft.Extensions.Configuration.UserSecrets -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Configuration.UserSecrets\netstandard2.0-Release\Microsoft.Extensions.Configuration.UserSecrets.dll
  System.Data.OleDb -> C:\W080621\net6_debugtypeembedded\artifacts\bin\System.Data.OleDb\net6.0-windows-Release\System.Data.OleDb.dll
  Microsoft.Extensions.Logging.Console -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Logging.Console\netstandard2.0-Release\Microsoft.Extensions.Logging.Console.dll
  Microsoft.Extensions.Logging.EventSource -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Logging.EventSource\net6.0-Release\Microsoft.Extensions.Logging.EventSource.dll
  System.DirectoryServices.AccountManagement -> C:\W080621\net6_debugtypeembedded\artifacts\bin\System.DirectoryServices.AccountManagement\net6.0-windows-Release\System.DirectoryServices.AccountManagement.dll
  Microsoft.Extensions.Logging.Console -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Logging.Console\net6.0-Release\Microsoft.Extensions.Logging.Console.dll
  Microsoft.Extensions.Hosting -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Hosting\netstandard2.1-Release\Microsoft.Extensions.Hosting.dll
  System.Diagnostics.EventLog -> C:\W080621\net6_debugtypeembedded\artifacts\bin\System.Diagnostics.EventLog\net6.0-Release\System.Diagnostics.EventLog.dll
  Microsoft.Extensions.Hosting.WindowsServices -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Hosting.WindowsServices\netstandard2.1-Release\Microsoft.Extensions.Hosting.WindowsServices.dll
  Microsoft.Extensions.Hosting.Systemd -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Hosting.Systemd\netstandard2.1-Release\Microsoft.Extensions.Hosting.Systemd.dll
  Microsoft.Extensions.Hosting -> C:\W080621\net6_debugtypeembedded\artifacts\bin\Microsoft.Extensions.Hosting\net6.0-Release\Microsoft.Extensions.Hosting.dll
  mscorlib -> C:\W080621\net6_debugtypeembedded\artifacts\bin\manual.mscorlib\net6.0-Release\mscorlib.dll
  System -> C:\W080621\net6_debugtypeembedded\artifacts\bin\manual.System\net6.0-Release\System.dll
  System.Data -> C:\W080621\net6_debugtypeembedded\artifacts\bin\manual.System.Data\net6.0-Release\System.Data.dll
  System.Xml -> C:\W080621\net6_debugtypeembedded\artifacts\bin\manual.System.Xml\net6.0-Release\System.Xml.dll
  ApiCompat ->
  Trimming win-x64 runtime pack assemblies with ILLinker...
  Trimming win-x64 OOB assemblies with ILLinker...
C:\Program Files\dotnet\sdk\6.0.100-preview.6.21355.2\Microsoft.Common.CurrentVersion.targets(4683,5): error MSB3030: Could not copy the file "C:\W080621\net6_debugtypeembedded\artifacts\bin\coreclr\windows.x64.Release\PDB\System.Private.CoreLib.pdb" because it was not found. [C:\W080621\net6_debugtypeembedded\src\libraries\externals.csproj]

Build FAILED.

C:\Program Files\dotnet\sdk\6.0.100-preview.6.21355.2\Microsoft.Common.CurrentVersion.targets(4683,5): error MSB3030: Could not copy the file "C:\W080621\net6_debugtypeembedded\artifacts\bin\coreclr\windows.x64.Release\PDB\System.Private.CoreLib.pdb" because it was not found. [C:\W080621\net6_debugtypeembedded\src\libraries\externals.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:07:24.45
Build failed with exit code 1. Check errors above.
Some builds failed:
        Configuration: Release, Architecture: x64
PS C:\W080621\net6_debugtypeembedded>
```

### Configuration

Windows 10, x64:

### Regression?

Not sure, I haven't tried building with `/p:DebugType=embedded` for previous .NET versions.

### Other information

I've also tried `/p:DebugType=` with `full`, `pdbonly`, and `portable` which in all three cases build successfully. Just having trouble with `embedded`

Thanks

**Comment by L2** [comment]
I was hoping to use /p:DebugType=embedded to see if the pdb information for the runtime libraries and such could be embedded inside the dlls instead of separate pdb files.

**Comment by danmoseley** [comment]
This will fix it. But, I don't really know what's supposed to happen with embedded PDB's with linking, etc.
```
C:\git\runtime>git diff
diff --git a/eng/liveBuilds.targets b/eng/liveBuilds.targets
index 7710f9d5e2f..d62f4bcd438 100644
--- a/eng/liveBuilds.targets
+++ b/eng/liveBuilds.targets
@@ -87,7 +87,7 @@
           $(CoreCLRSharedFrameworkPdbDir)*.pdb;
           $(CoreCLRSharedFrameworkPdbDir)*.dbg;
           $(CoreCLRSharedFrameworkPdbDir)*.dwarf" IsNative="true" />
-      <RuntimeFiles
+      <RuntimeFiles Condition="Exists('$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.pdb')"
         Include="$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.pdb" />
       <RuntimeFiles Condition="Exists('$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.ni.pdb')"
         Include="$(CoreCLRArtifactsPdbDir)System.Private.CoreLib.ni.pdb" />
```

**Comment by danmoseley** [comment]
@L2 maybe try that edit locally and let us know if everything seems to work (not just the build but debugging eg)

**Comment by L2** [comment]
Thanks for the quick help @danmoseley , everything's going well after your fix :)

**Comment by ViktorHofer** [comment]
@L2 would you mind submitting a PR to fix this upstream as well?

**Comment by L2** [comment]
Sure @ViktorHofer , the fix is from @danmoseley . Just wanted to double check with you @danmoseley , do you want to submit your fix upstream or should I? Thanks.

**Comment by danmoseley** [comment]
Do you mean main, @ViktorHofer ? @L2 yes, I would welcome a PR into main -- thank you!

**Comment by danmoseley** [comment]
Again, I've no idea if the build ends up doing something useful, but this seems like a reasonable change anyway.

@agocke would embedded PDB's cause a problem with trimming?

**Comment by agocke** [comment]
It should be fine -- embedded resources should be preserved during trimming.

**Comment by agocke** [comment]
However, I don't necessarily think it's a good idea to ship with embedded PDBs, just that for local builds, it shouldn't be a problem.

**Comment by L2** [comment]
Thanks @danmoseley and @agocke . Added this fix to a PR (#57161)

**Comment by danmoseley** [comment]
@agocke ah, for some reason I thought there were circumstances in which the trimming edits the PDB's, but perhaps that's only when linking. thx
